### PR TITLE
Register ECR credential plugin only when  AWS cloud instance is created

### PR DESCRIPTION
**Pull request by therc**
Try to fix fallout from #19447.

### CODE REVIEW

**Comment by erictune** [comment]
This should fix it.

File: pkg/credentialprovider/aws/aws_credentials.go [hunk 1]
```diff
@@ -66,9 +65,9 @@ type ecrProvider struct {
 	getter tokenGetter
 }
 
-// init registers the various means by which ECR credentials may
+// Init registers the various means by which ECR credentials may
 // be resolved.
-func init() {
+func Init() {
```
Comment (erictune) [inline pkg/credentialprovider/aws/aws_credentials.go]: Add a comment above saying something like "// Not using the package init() function: this module should be initialized only if using the AWS cloud provider.  This avoids timeout waiting for non-existent provider."

**Comment by therc** [comment]
There's still the risk of registering the credential provider twice and logging a fatal message, I think, if newAWSCloud is called multiple times. Which would be fixed, in turn, by using a package-wide bool telling us we've already done it.

**Comment by therc** [comment]
I now call Init() just once, in a thread-safe manner, with the help of sync.Once.

**Comment by mikedanese** [comment]
Fixes people's ./hack/test-cmd.sh so i'm merging.

**Pull request merged**